server/app/controllers/admin/AdminQuestionTranslationsController.java

```diff
@@ -30,17 +35,41 @@ public class AdminQuestionTranslationsController extends CiviFormController {
   private final QuestionService questionService;
   private final QuestionTranslationView translationView;
   private final FormFactory formFactory;
+  private final TranslationLocales translationLocales;
+  private final Optional<Locale> maybeFirstLocaleForTranslations;
 
   @Inject
   public AdminQuestionTranslationsController(
       HttpExecutionContext httpExecutionContext,
       QuestionService questionService,
       QuestionTranslationView translationView,
-      FormFactory formFactory) {
-    this.httpExecutionContext = httpExecutionContext;
-    this.questionService = questionService;
-    this.translationView = translationView;
-    this.formFactory = formFactory;
+      FormFactory formFactory,
+      TranslationLocales translationLocales) {
+    this.httpExecutionContext = checkNotNull(httpExecutionContext);
+    this.questionService = checkNotNull(questionService);
+    this.translationView = checkNotNull(translationView);
+    this.formFactory = checkNotNull(formFactory);
+    this.translationLocales = checkNotNull(translationLocales);
+    this.maybeFirstLocaleForTranslations =
+        this.translationLocales.localesForTranslation().stream().findFirst();
+  }
+
+  /**
+   * Redirects to the first non-English locale eligible for translations for a a program.
+   *
+   * @param request the current {@link Http.Request}
+   * @param id the ID of the program to update
+   */
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result redirectToFirstLocale(Http.Request request, long id) {
+    if (maybeFirstLocaleForTranslations.isEmpty()) {
+      return redirect(routes.AdminQuestionController.index().url())
+          .flashing("error", "Translations are not enabled for this configuration");
```

Comment:
Question: should toast messages be sentence structure?  EG there isn't punctuation here.